### ISSUE TITLE
AppPath any other but ${workspaceRoot} freezes

### ISSUE THREAD

**Issue by Swiftwork** [NativeScript/nativescript-vscode-extension#104]
When appRoot isn't actually the workspace root but rather a sub directory "${workspaceRoot}/nativescript" the plugin won't get past the initialization call due to the [server creating a socket hash from the workspaceRoot and not appRoot](https://github.com/NativeScript/nativescript-vscode-extension/blob/master/src/ipc/extensionServer.ts#L22-L24)

**Comment by ivanbuhov** [comment]
@Swiftwork You are absolutely right. We have to think about another mechanism for calculating the unix socket path used for communication between the debug adapter and the extension host process.

**Comment by manojdcoder** [comment]
👍

**Comment by manojdcoder** [comment]
+1

**Comment by Gyunikuchan** [comment]
+1

**Comment by stylefish** [comment]
anything new on this? i'm trying to develop a nativescript + electron desktop app with some code share and cannot let nativescript live inside the workspace folder. for now i'm using console.log() debugging ;)
so it would be nice to have this ;)
thanks for feedback!

**Comment by madmath03** [comment]
+1

**Comment by harrijon** [comment]
+1

**Comment by 5te1n** [comment]
I can't debug my plugin-seed, because of this issue. Is there already a fix for this?
